authorization/storage_authorization.go

```diff
@@ -141,6 +182,27 @@ func (s *Store) GetAuthorizationByID(ctx context.Context, tx kv.Tx, id platform.
 	return a, nil
 }
 
+// validateToken checks if token matches that token stored in auth. If auth.Token is set, that is
+// compared first. Otherwise, auth.HashedToken is used to verify token. If neither field in auth is set, then
+// the comparison fails.
+func (s *Store) validateToken(auth *influxdb.Authorization, token string) (bool, error) {
+	if auth.Token != "" {
+		return auth.Token == token, nil
+	}
+
+	if auth.HashedToken != "" {
+		match, err := s.hasher.Match(auth.HashedToken, token)
+		if err != nil {
+			return false, fmt.Errorf("error matching hashed token for validation: %w", err)
```

Comment:
More information like ID and description?

Reply:
Added.